### ISSUE TITLE
error in mutation spikein while running test_snv.py

### ISSUE THREAD

**Issue by rbatorsky-claritas** [adamewing/bamsurgeon#74]
I am running the following:
 ./test_snv_bwamem.sh 1 1 ../test_data/Homo_sapiens_chr22_assembly19.fasta ~/BioSoftware/picard-tools-1.134/picard.jar 

It seems that some of the samtools steps are running correctly and some are not. I am running pysam 0.9.2 and have verified all dependencies are installed and on my path.

The result is:
INFO    2016-10-17 15:44:04.542162  starting /usr/local/bin/addsnv.py called with args: /usr/local/bin/addsnv.py -v ../test_data/random_snvs.txt -f ../test_data/testregion_realign.bam -r ../test_data/Homo_sapiens_chr22_assembly19.fasta -o ../test_data/testregion_mut.bam -n 1 -c ../test_data/test_cnvlist.txt.gz -p 1 --picardjar /Users/rebeccabatorsky/BioSoftware/picard-tools-1.134/picard.jar --aligner mem --seed 1234

INFO    2016-10-17 15:44:04.543324  created tmp directory: addsnv.tmp
INFO    2016-10-17 15:44:04.543457  created directory: addsnv_logs_testregion_mut.bam
Debug, haploclusters:[[{'start': 34166720, 'end': 34166721, 'chrom': '22', 'altbase': None, 'vaf': 1.0}]]
INFO    2016-10-17 15:44:04.555897  haplo_22_34166721_34166721  creating tmp bam:  addsnv.tmp/haplo_22_34166721_34166721.tmpbam.735cd520-a5e4-4d4b-98b3-b145819813ab.bam
INFO    2016-10-17 15:44:05.746773  haplo_22_34166721_34166721  len(readlist): 58
INFO    2016-10-17 15:44:05.747037  haplo_22_34166721_34166721  copy number in snp region: 22 34166721 34166721 = 3.0

adjusted VAF: 0.333333333333

INFO    2016-10-17 15:44:05.747269  haplo_22_34166721_34166721  picked: 19
INFO    2016-10-17 15:44:05.748240  haplo_22_34166721_34166721  wrote:  58 mutated: 19
INFO    2016-10-17 15:44:05.751673  haplo_22_34166721_34166721  converting addsnv.tmp/haplo_22_34166721_34166721.tmpbam.735cd520-a5e4-4d4b-98b3-b145819813ab.bam to fastq

INFO    2016-10-17 15:44:05.752133  converting BAM addsnv.tmp/haplo_22_34166721_34166721.tmpbam.735cd520-a5e4-4d4b-98b3-b145819813ab.bam to FASTQ
[Mon Oct 17 15:44:06 EDT 2016] picard.sam.SamToFastq INPUT=addsnv.tmp/haplo_22_34166721_34166721.tmpbam.735cd520-a5e4-4d4b-98b3-b145819813ab.bam FASTQ=addsnv.tmp/haplo_22_34166721_34166721.tmpbam.735cd520-a5e4-4d4b-98b3-b145819813ab.fastq INTERLEAVE=true INCLUDE_NON_PRIMARY_ALIGNMENTS=false VALIDATION_STRINGENCY=SILENT    OUTPUT_PER_RG=false RG_TAG=PU RE_REVERSE=true INCLUDE_NON_PF_READS=false READ1_TRIM=0 READ2_TRIM=0 VERBOSITY=INFO QUIET=false COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json
[Mon Oct 17 15:44:06 EDT 2016] Executing as rebeccabatorsky@CGPVTLAP0050.local on Mac OS X 10.10.5 x86_64; Java HotSpot(TM) 64-Bit Server VM 1.8.0_51-b16; Picard version: 1.134(a7a08c474e4d99346eec7a9956a8fe71943b5d80_1434033355) JdkDeflater
[Mon Oct 17 15:44:06 EDT 2016] picard.sam.SamToFastq done. Elapsed time: 0.00 minutes.
Runtime.totalMemory()=257425408
INFO    2016-10-17 15:44:06.088677  haplo_22_34166721_34166721  aligning addsnv.tmp/haplo_22_34166721_34166721.tmpbam.735cd520-a5e4-4d4b-98b3-b145819813ab.fastq with bwa mem

[M::bwa_idx_load_from_disk] read 0 ALT contigs
[M::process] read 116 sequences (11600 bp)...
[M::process] 0 single-end sequences; 116 paired-end sequences
[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 58, 0, 0)
[M::mem_pestat] skip orientation FF as there are not enough pairs
[M::mem_pestat] analyzing insert size distribution for orientation FR...
[M::mem_pestat](25, 50, 75) percentile: (469, 501, 535)
[M::mem_pestat] low and high boundaries for computing mean and std.dev: (337, 667)
[M::mem_pestat] mean and std.dev: (502.26, 51.98)
[M::mem_pestat] low and high boundaries for proper pairs: (271, 733)
[M::mem_pestat] skip orientation RF as there are not enough pairs
[M::mem_pestat] skip orientation RR as there are not enough pairs
[M::mem_process_seqs] Processed 116 reads in 0.006 CPU sec, 0.006 real sec
[main] Version: 0.7.12-r1039
[main] CMD: bwa mem -t 1 -M -p ../test_data/Homo_sapiens_chr22_assembly19.fasta addsnv.tmp/haplo_22_34166721_34166721.tmpbam.735cd520-a5e4-4d4b-98b3-b145819813ab.fastq
[main] Real time: 0.058 sec; CPU: 0.060 sec
INFO    2016-10-17 15:44:06.150695  haplo_22_34166721_34166721  writing addsnv.tmp/haplo_22_34166721_34166721.tmpbam.735cd520-a5e4-4d4b-98b3-b145819813ab.bam.realign.sam to BAM...
INFO    2016-10-17 15:44:06.156371  haplo_22_34166721_34166721  deleting SAM: addsnv.tmp/haplo_22_34166721_34166721.tmpbam.735cd520-a5e4-4d4b-98b3-b145819813ab.bam.realign.sam
INFO    2016-10-17 15:44:06.156536  haplo_22_34166721_34166721  sorting output: samtools sort -@ 1 -m 10000000000 -T addsnv.tmp/haplo_22_34166721_34166721.tmpbam.735cd520-a5e4-4d4b-98b3-b145819813ab.bam.realign.sorted.bam -o addsnv.tmp/haplo_22_34166721_34166721.tmpbam.735cd520-a5e4-4d4b-98b3-b145819813ab.bam.realign.sorted.bam addsnv.tmp/haplo_22_34166721_34166721.tmpbam.735cd520-a5e4-4d4b-98b3-b145819813ab.bam
INFO    2016-10-17 15:44:06.161041  haplo_22_34166721_34166721  remove original bam:addsnv.tmp/haplo_22_34166721_34166721.tmpbam.735cd520-a5e4-4d4b-98b3-b145819813ab.bam
INFO    2016-10-17 15:44:06.161194  haplo_22_34166721_34166721  rename sorted bam: addsnv.tmp/haplo_22_34166721_34166721.tmpbam.735cd520-a5e4-4d4b-98b3-b145819813ab.bam.realign.sorted.bam to original name: addsnv.tmp/haplo_22_34166721_34166721.tmpbam.735cd520-a5e4-4d4b-98b3-b145819813ab.bam
INFO    2016-10-17 15:44:06.161334  haplo_22_34166721_34166721  indexing: samtools index addsnv.tmp/haplo_22_34166721_34166721.tmpbam.735cd520-a5e4-4d4b-98b3-b145819813ab.bam
[bam_index] corrupted or unsorted input file
INFO    2016-10-17 15:44:06.165385  haplo_22_34166721_34166721  removing addsnv.tmp/haplo_22_34166721_34166721.tmpbam.735cd520-a5e4-4d4b-98b3-b145819813ab.fastq

---

ERROR   2016-10-17 15:44:06.167530  encountered error in mutation spikein: ['22_34166720_34166721_1.0_None']
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/bamsurgeon-1.0-py2.7.egg/EGG-INFO/scripts/addsnv.py", line 280, in makemut
  File "/Library/Python/2.7/site-packages/bamsurgeon-1.0-py2.7.egg/EGG-INFO/scripts/addsnv.py", line 56, in countReadCoverage
  File "pysam/calignmentfile.pyx", line 1116, in pysam.calignmentfile.AlignmentFile.pileup (pysam/calignmentfile.c:12769)
NotImplementedError: pileup of samfiles not implemented yet

---

INFO    2016-10-17 15:44:06.168823  no succesful mutations
Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid
[E::hts_open_format] fail to open file '../test_data/testregion_mut.bam'
[mpileup] failed to open ../test_data/testregion_mut.bam: No such file or directory
Failed to open -: unknown file type

**Comment by adamewing** [comment]
Hi,

"NotImplementedError: pileup of samfiles not implemented yet" is a new one for me ... it seems to be coming from an earlier samtools error after the reads are realigned. Could you let me know what version of samtools you've got installed? Also, I think I noticed in another thread you mentioned changing something to shell=True - if that was the case / still is the case could you let me know what lines you changed? (sorry if I'm misremembering)

Thanks!

**Comment by rbatorsky-claritas** [comment]
Thanks for your reply.

I am using samtools 1.2-187-g3e9357c, bwa 0.7.12-r1044, pysam 0.9.2

In an earlier thread I did mention needing to pass shell=True to
subprocess.call in order to properly view errors coming from the subprocess
in the terminal but after I went through and installed updated versions of
all requirements and reinstalled bamsurgeon, I no longer needed that
change. The code that generates this error does not have any modifications.
